File: Assets/UnitySerializer/Scripts/Movement.js
#pragma strict


private var controller :CharacterController;
controller = gameObject.GetComponent(CharacterController);

public var texture : Texture2D;
 
private var moveDirection = Vector3.zero;
private var forward = Vector3.zero;
private var right = Vector3.zero;

function Start () {
	texture = null;
	
}

function Update () {
    forward = transform.forward;
	right = Vector3(forward.z, 0, -forward.x);
 
	var horizontalInput = Input.GetAxisRaw("Horizontal");
	var verticalInput = Input.GetAxisRaw("Vertical");
	var targetDirection = horizontalInput * right + verticalInput * forward;	
 
	 moveDirection = Vector3.RotateTowards(moveDirection, targetDirection, 200 * Mathf.Deg2Rad * Time.deltaTime, 1000);
   
	var movement = moveDirection  * Time.deltaTime * 10;
	controller.Move(movement);
}

File: Assets/UnitySerializerExamples/Scripts/Movement.cs
using UnityEngine;

public class Movement : MonoBehaviour {
    private CharacterController controller;

    private Vector3 moveDirection = Vector3.zero;
    private Vector3 forward = Vector3.zero;
    private Vector3 right = Vector3.zero;

    private void Start() {
        controller = GetComponent<CharacterController>();
    }

    private void Update() {
        forward = GetComponent<Transform>().forward;
        right = new Vector3(forward.z, 0.0f, -forward.x);

        float horizontalInput = Input.GetAxisRaw("Horizontal");
        float verticalInput = Input.GetAxisRaw("Vertical");
        Vector3 targetDirection = horizontalInput * right + verticalInput * forward;

        moveDirection = Vector3.RotateTowards(moveDirection, targetDirection, 200.0f * Mathf.Deg2Rad * Time.deltaTime, 1000.0f);

        Vector3 movement = moveDirection * Time.deltaTime * 10.0f;
        controller.Move(movement);
    }
}
